commit fb9ff398da9f772bc4a73143a67bb71bbf5b45fa
Author: Matthias Reichl <hias@horus.com>
Date:   Fri Feb 2 20:30:42 2018 +0100

    ASoC: allo-boss-dac: fix S24_LE format
    
    Remove set_bclk_ratio call so 24-bit data is transmitted in
    24 bclk cycles.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/bcm/allo-boss-dac.c b/sound/soc/bcm/allo-boss-dac.c
index 203ab76c7045..26c37da07017 100644
--- a/sound/soc/bcm/allo-boss-dac.c
+++ b/sound/soc/bcm/allo-boss-dac.c
@@ -222,14 +222,6 @@ static int snd_allo_boss_update_rate_den(
 	return 0;
 }
 
-static int snd_allo_boss_set_bclk_ratio_pro(
-	struct snd_soc_dai *cpu_dai, struct snd_pcm_hw_params *params)
-{
-	int bratio = snd_pcm_format_physical_width(params_format(params))
-		* params_channels(params);
-	return snd_soc_dai_set_bclk_ratio(cpu_dai, bratio);
-}
-
 static void snd_allo_boss_gpio_mute(struct snd_soc_card *card)
 {
 	if (mute_gpio)
@@ -281,9 +273,6 @@ static int snd_allo_boss_hw_params(
 {
 	int ret = 0;
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-	unsigned int sample_bits =
-		snd_pcm_format_physical_width(params_format(params));
 
 	if (snd_soc_allo_boss_master) {
 		struct snd_soc_codec *codec = rtd->codec;
@@ -291,13 +280,8 @@ static int snd_allo_boss_hw_params(
 		snd_allo_boss_set_sclk(codec,
 			params_rate(params));
 
-		ret = snd_allo_boss_set_bclk_ratio_pro(cpu_dai,
-			params);
-		if (!ret)
-			ret = snd_allo_boss_update_rate_den(
-				substream, params);
-	} else {
-		ret = snd_soc_dai_set_bclk_ratio(cpu_dai, sample_bits * 2);
+		ret = snd_allo_boss_update_rate_den(
+			substream, params);
 	}
 	return ret;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1188_fb9ff398da9f772bc4a73143a67bb71bbf5b45fa.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1188_fb9ff398da9f772bc4a73143a67bb71bbf5b45fa.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
